src/effects/effectchainslot.h

```diff
@@ -44,73 +48,82 @@ class EffectChainSlot : public QObject {
-                      unsigned int slotNumber);
+    // Get the human-readable name of the EffectChain
+    const QString& name() const;
+    void setName(const QString& name);
+
+    // Get the human-readable description of the EffectChain
+    QString description() const;
+    void setDescription(const QString& description);
+
+    static QString mixModeToString(EffectChainMixMode type) {
+        switch (type) {
+            case EffectChainMixMode::DrySlashWet:
+                return "DRY/WET";
+            case EffectChainMixMode::DryPlusWet:
+                return "DRY+WET";
+            default:
+                return "UNKNOWN";
+        }
+    }
+    static EffectChainMixMode mixModeFromString(const QString& typeStr) {
+        if (typeStr == "DRY/WET") {
+            return EffectChainMixMode::DrySlashWet;
+        } else if (typeStr == "DRY+WET") {
+            return EffectChainMixMode::DryPlusWet;
+        } else {
+            return EffectChainMixMode::NumMixModes;
+        }
+    }
 
-    // Indicates that the given EffectChain was loaded into this
-    // EffectChainSlot
-    void effectChainLoaded(EffectChainPointer pEffectChain);
+    void loadEffect(const unsigned int iEffectSlotNumber,
+                    const QString& id, EffectManifestPointer pManifest,
+                    EffectInstantiatorPointer pInstantiator);
+    void removeEffect(unsigned int effectSlotNumber);
 
+  signals:
     // Signal that whoever is in charge of this EffectChainSlot should load the
     // next EffectChain into it.
     void nextChain(unsigned int iChainSlotNumber,
-                   EffectChainPointer pEffectChain);
+                   EffectChainSlotPointer pEffectChain);
 
     // Signal that whoever is in charge of this EffectChainSlot should load the
     // previous EffectChain into it.
     void prevChain(unsigned int iChainSlotNumber,
-                   EffectChainPointer pEffectChain);
+                   EffectChainSlotPointer pEffectChain);
 
     // Signal that whoever is in charge of this EffectChainSlot should clear
     // this EffectChain (by removing the chain from this EffectChainSlot).
-    void clearChain(unsigned int iChainNumber, EffectChainPointer pEffectChain);
-
-    // Signal that whoever is in charge of this EffectChainSlot should load the
-    // next Effect into the specified EffectSlot.
-    void nextEffect(unsigned int iChainSlotNumber,
-                    unsigned int iEffectSlotNumber,
-                    EffectPointer pEffect);
-
-    // Signal that whoever is in charge of this EffectChainSlot should load the
-    // previous Effect into the specified EffectSlot.
-    void prevEffect(unsigned int iChainSlotNumber,
-                    unsigned int iEffectSlotNumber,
-                    EffectPointer pEffect);
+    void clearChain(unsigned int iChainNumber, EffectChainSlotPointer pEffectChain);
 
     // Signal that indicates that the EffectChainSlot has been updated.
     void updated();
```

Comment:
This signal name is quite unhelpful. Change it to `effectLoaded`. Be sure to update WEffectSelector.

Reply:
I've set it to `effectChanged` because we are supposed to trigger this signal on unloading as well.